artifacttemplates/http%3A%2F%2Fopentosca.org%2Fartifacttemplates/OpenStack_CloudProviderInterfaceIA-w2/source/src/main/java/org/opentosca/artifacttemplates/OpenToscaHeaders.java

```diff
@@ -1,10 +1,29 @@
 package org.opentosca.artifacttemplates;
 
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import javax.xml.namespace.QName;
 
 public record OpenToscaHeaders(String messageId,
                                String replyTo,
-                               Map<QName, Map<String, String>> deploymentArtifactsMap) {
+                               Map<QName, Map<String, String>> deploymentArtifacts) {
+
+    @Override
+    public String toString() {
+        String daString = deploymentArtifacts.entrySet().stream()
```

Comment:
Not sure if this is the most readable solution...

Reply:
My concern was rather the readability of the toString method itself

Reply:
Update:

```
OpenToscaHeaders {
  messageId = 'DC357ED0DE243E5-0000000000000000',
  replyTo   = 'http://192.168.2.128:8087/callback',
  deploymentArtifacts = {
    '{http://opentosca.org/artifacttypes}ScriptArtifact': {
      'install': 'http://192.168.2.128:1337/csars/test_w1-wip1-selfContained-w1-wip1.csar/content/artifacttemplates/http%253A%252F%252Fopentosca.org%252Fartifacttemplates/SomeArtifact_w1/files/insall.sh'
    },
    '{http://opentosca.org/artifacttypes}CloudImage': {
      'focal-server-cloudimg-amd64.img': 'http://192.168.2.128:1337/csars/test_w1-wip1-selfContained-w1-wip1.csar/content/artifacttemplates/http%253A%252F%252Fopentosca.org%252Fartifacttemplates/Ubuntu-VM-DA_20.04-CloudImage-w1-wip1/files/focal-server-cloudimg-amd64.img',
      'test': 'asasdfjklasdjflkasjdflkasjdflkasjdflkasjdflkasjdflkjsadkljflksadjflkasdjflköasdjflk'
    }
  }
}
```